docs/en/stack/ml/nlp/ml-nlp-elastic-rerank.asciidoc

```diff
@@ -0,0 +1,242 @@
+[[ml-nlp-rerank]]
+= Elastic Rerank
+
+Elastic Rerank is a state-of-the-art cross-encoder reranking model trained by Elastic that helps you improve search relevance with a few simple API calls.
+Elastic Rerank is Elastic's first semantic reranking model and is available out-of-the-box in supporting Elastic deployments using the {es} {infer-cap} API.
+
+Use Elastic Rerank to improve existing search applications including:
+
+* Traditional BM25 scoring
+* Hybrid semantic search
+* Retrieval Augmented Generation (RAG)
+
+The model can significantly improve search result quality by reordering results based on deeper semantic understanding of queries and documents.
+
+When reranking BM25 results, it provides an average 40% improvement in ranking quality on a diverse benchmark of retrieval tasks— matching the performance of models 11x its size.
+
+[discrete]
+[[ml-nlp-rerank-availability]]
+== Availability and requirements 
+
+experimental[]
+
+[discrete]
+[[ml-nlp-rerank-availability-serverless]]
+=== Elastic Cloud Serverless
+
+Elastic Rerank is available in Elasticsearch Serverless projects as of November 25, 2024.
+
+[discrete]
+[[ml-nlp-rerank-availability-elastic-stack]]
+=== Elastic Cloud Hosted and self-managed deployments
+
+Elastic Rerank is available in Elastic Stack version 8.17+:
+
+* To use Elastic Rerank, you must have the appropriate subscription level or the trial period activated.
+* A 4GB ML node
++
+[IMPORTANT]
+====
+Deploying the Elastic Rerank model in combination with ELSER (or other hosted models) requires at minimum an 8GB ML node. The maximum size for trial ML nodes is 4GB (defaults to 1GB).
+====
+
+[discrete]
+[[ml-nlp-rerank-deploy]]
+== Download and deploy
+
+To download and deploy Elastic Rerank, use the {ref}/infer-service-elasticsearch.html[create inference API] to create an Elasticsearch service `rerank` endpoint.
+
+[discrete]
+[[ml-nlp-rerank-deploy-steps]]
+=== Create an inference endpoint
+
+. In {kib}, navigate to the *Dev Console*.
+
+. Create an {infer} endpoint with the Elastic Rerank service by running:
++
+[source,console]
+----------------------------------
+PUT _inference/text_similarity/my-rerank-model
+{
+  "service": "elasticsearch",
+  "service_settings": {
+    "adaptive_allocations": {
+      "enabled": true,
+      "min_number_of_allocations": 1,
+      "max_number_of_allocations": 10
+    },
+    "num_threads": 1,
+    "model_id": ".elastic-rerank"
+  }
+}
+----------------------------------
++
+NOTE: The API request automatically downloads and deploys the model. This example uses <<ml-nlp-auto-scale,autoscaling>> through adaptive allocation.
```

Comment:
We know this command may timeout in kibana with the bad gateway. Istvan added a note to the inference API docs, is it worth repeating that here

https://github.com/elastic/elasticsearch/blob/c2bd0de27801224c11f85809c78c6847652662b2/docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc?plain=1#L55

Reply:
Oh yes good point 👍